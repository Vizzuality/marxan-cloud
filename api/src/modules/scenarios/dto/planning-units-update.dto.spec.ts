import { validateSync } from 'class-validator';
import {
  getDtoWithInvalidUuids,
  getDtoWithInvalidMultiPolygon,
} from './__mocks__/dtos.data';

describe(`when changing by id`, () => {
  describe(`when providing invalid uuid`, () => {
    it(`should return error`, () => {
      expect(
        validateSync(getDtoWithInvalidUuids())[0].children,
      ).toMatchSnapshot();
    });
  });
});

describe(`when changing by geo`, () => {
  describe(`when providing invalid shape`, () => {
    it(`should return error`, () => {
      expect(
        validateSync(getDtoWithInvalidMultiPolygon())[0].children,
      ).toMatchSnapshot();
    });
  });
});
