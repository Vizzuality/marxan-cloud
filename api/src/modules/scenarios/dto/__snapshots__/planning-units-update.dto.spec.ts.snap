// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when changing by geo when providing invalid shape should return error 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "IsMultiPolygon": "Value must be a valid Geometry(MultiPolygon)",
    },
    "property": "exclude",
    "target": PlanningUnitsByGeoJsonUpdateDto {
      "exclude": Object {
        "coordinates": Array [
          Array [
            Array [
              Array [
                102,
                2,
              ],
              Array [
                104,
                5,
              ],
              Array [
                103,
                3,
              ],
            ],
          ],
        ],
        "type": "MultiPolygon",
      },
      "include": Object {
        "coordinates": Array [
          Array [
            Array [
              Array [
                102,
                2,
              ],
              Array [
                103,
                2,
              ],
              Array [
                103,
                3,
              ],
              Array [
                102,
                3,
              ],
              Array [
                102,
                2,
              ],
            ],
          ],
          Array [
            Array [
              Array [
                100,
                0,
              ],
              Array [
                101,
                0,
              ],
              Array [
                101,
                1,
              ],
              Array [
                100,
                1,
              ],
              Array [
                100,
                0,
              ],
            ],
            Array [
              Array [
                100.2,
                0.2,
              ],
              Array [
                100.8,
                0.2,
              ],
              Array [
                100.8,
                0.8,
              ],
              Array [
                100.2,
                0.8,
              ],
              Array [
                100.2,
                0.2,
              ],
            ],
          ],
        ],
        "type": "MultiPolygon",
      },
    },
    "value": Object {
      "coordinates": Array [
        Array [
          Array [
            Array [
              102,
              2,
            ],
            Array [
              104,
              5,
            ],
            Array [
              103,
              3,
            ],
          ],
        ],
      ],
      "type": "MultiPolygon",
    },
  },
]
`;

exports[`when changing by id when providing invalid uuid should return error 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isUuid": "each value in include must be an UUID",
    },
    "property": "include",
    "target": PlanningUnitsByIdUpdateDto {
      "exclude": Array [
        "ab957e1d-0271-4571-b92c-19d2efec0f75",
      ],
      "include": Array [
        "non-uuid",
      ],
    },
    "value": Array [
      "non-uuid",
    ],
  },
]
`;
